jax/_src/scipy/stats/beta.py

```diff
@@ -59,12 +59,17 @@ def cdf(x: ArrayLike, a: ArrayLike, b: ArrayLike,
 
 @_wraps(osp_stats.beta.logcdf, update_doc=False)
 def logcdf(x: ArrayLike, a: ArrayLike, b: ArrayLike,
-        loc: ArrayLike = 0, scale: ArrayLike = 1) -> Array:
+           loc: ArrayLike = 0, scale: ArrayLike = 1) -> Array:
   return lax.log(cdf(x, a, b, loc, scale))
 
 
 @_wraps(osp_stats.beta.sf, update_doc=False)
 def sf(x: ArrayLike, a: ArrayLike, b: ArrayLike,
-        loc: ArrayLike = 0, scale: ArrayLike = 1) -> Array:
-  cdf_result = cdf(x, a, b, loc, scale)
-  return lax.sub(_lax_const(cdf_result, 1), cdf_result)
+       loc: ArrayLike = 0, scale: ArrayLike = 1) -> Array:
+  return cdf(2 * loc + scale - x, b, a, loc, scale)
```

Comment:
I don't know much about the beta distribution, but why is this correct? It looks to me like you want to compute the complement of the incomplete beta function here, but why is this it?

Reply:
Exactly: the complement of `betainc(a, b, x)` is `betainc(b, a, 1 - x)`. This computation is effectively doing this, except it must take into account that in this function the endpoints are generalized from `(0, 1)` to `(loc, loc + scale)`.

So we end up with `betainc(b, a, loc + scale - (x - loc))`, which simplifies to `betainc(b, a, 2 * loc + scale - x)`.

Reply:
I ended up rewriting it in terms of betainc directly – I think it's more clear this way.